### ISSUE TITLE
cursor iterator only works for K: From<&u[u8]>

### ISSUE THREAD

**Issue by djmitche** [zshipko/rust-kv#10]
This seems, at least for `K = &str`, to make it impossible to iterate over a cursor.

To reproduce, add the following to the main example in [`src/lib.rs`](https://github.com/zshipko/rust-kv/blob/9972460aaba77327a8d5356db8944a7f9d2715aa/src/lib.rs#L44):

```rust
    {
        // This time a readonly transaction
        let txn = store.read_txn()?;

        // Getting a value is easy once everything is set up
        let curs = txn.read_cursor(&bucket)?;
        let all: Vec<(String, String)> =
            curs.map(|(k, v)| (k.to_string(), v.to_string())).collect();
    }
```
and the resulting error is
```
error[E0599]: no method named `map` found for type `kv::cursor::Cursor<'_, &str, &str>` in the current scope
  --> src/bin/foo.rs:44:18
   |
44 |             curs.map(|(k, v)| (k.to_string(), v.to_string())).collect();
   |                  ^^^ method not found in `kv::cursor::Cursor<'_, &str, &str>`
   |
   = note: the method `map` exists but the following trait bounds were not satisfied:
           `&mut kv::cursor::Cursor<'_, &str, &str> : std::iter::Iterator`
```
due to [this trait bound](https://github.com/zshipko/rust-kv/blob/9972460aaba77327a8d5356db8944a7f9d2715aa/src/cursor.rs#L56-L58)
```rust
impl<'a, K: Key, V: Value<'a>> Iterator for Iter<'a, K, V>
where
    K: From<&'a [u8]>,
```

**Comment by zshipko** [comment]
Thanks!

I've added your example here: https://github.com/zshipko/rust-kv/blob/master/src/lib.rs#L54

To enable this:
- I added `FromRawKey` trait to allow better control of key conversion 
- Added `Iterator` implementation for `Cursor`

Try using the `master` branch and let me know if this works for your use case!

**Comment by djmitche** [comment]
Very nice -- yes, that's great!  Thank you!!

I see from the source that the difference between `iter` and `iter_current` is that the latter starts at the current cursor location, whereas the former always starts at the beginning.  It might be worth adding a bit to the docs there for that purpose.  I can make a PR to do that if you'd like.

**Comment by zshipko** [comment]
Glad to hear! I will make sure to update that documentation before making the next release.

**Comment by zshipko** [comment]
Just released [0.11.0](https://crates.io/crates/kv/0.11.0). Thanks again for reporting this issue!
